src/stringClass.js

```diff
@@ -0,0 +1,162 @@
+const StringManipulation = {
+  /**
+   * Tests if a string contains vowels
+   * @return {Boolean} True or false depending on the match
+   */
+  hasVowels() {
+    return /[aeiou]/i.test(this);
+  },
+
+  /**
+   * Converts a string to uppercase characters
+   * @return {String} Word in uppercase
+   */
+  toUpper() {
+    return this.replace(/[a-z]/g, character =>
+      String.fromCharCode(character.charCodeAt() - 32)
+    );
+  },
+
+  /**
+   * Converts a string to lowercase characters
+   * @return {String} Word in lowercase
+   */
+  toLower() {
+    return this.replace(/[A-Z]/g, character =>
+      String.fromCharCode(character.charCodeAt() + 32)
+    );
+  },
+
+  /**
+   * Converts the first letter of a string to uppercase
+   * @return {String} Word with capitalized first letter
+   */
+  ucFirst() {
+    return this.replace(this[0], this[0].toUpper());
+  },
+
+  /**
+   * Tests if a string is a question
+   * @return {Boolean} True or false depending on the match
+   */
+  isQuestion() {
+    return /\?$/.test(this);
+  },
+
+  /**
+   *Creates a list of words from the string
+   @return {Array} A list of the words
+   */
+  words() {
+    return this.replace(/[^\w\s]/g, '').split(/[\s]/);
+  },
+
+  /**
+   * Checks for the number of words in a string
+   * @return {Number} The total number of words in the string
+   */
+  wordCount() {
+    return this.words().length;
+  },
+
+  /**
+   * Perfoms a reverse on a string
+   * @return {String} The reverse of a string
+   */
+  reverseWord() {
+    return this.replace(/./g, (letter, index) =>
+      this[this.length - (index + 1)]);
+  },
+
+  /**
+   * Performs a currency representation of a string
+   * @return {String} Represented as a currency
+   */
+  toCurrency() {
+    const currencyParts = this.split(/\./);
+    currencyParts[1] = currencyParts[1] || '00';
+    currencyParts[0] = currencyParts[0].reverseWord()
+      .replace(/.{3}/g, number => `${number},`)
+      .replace(/,$/, '')
+      .reverseWord();
+    return `${currencyParts[0]}.${currencyParts[1].substr(0, 2)}`;
+  },
+
+  /**
+   * Performs a number representation of a currency
+   * @return {String} Represented as a number
+   */
+  fromCurrency() {
+    return this.replace(/,/g, '');
```

Comment:
this may require you to strip away insignificant zeroes from input, e.g 1,234,567.00 -> 1234567
unless, no specific requirements was mentioned in the checkpoint